chart.js.d.ts

```diff
@@ -0,0 +1,605 @@
+      hitRadius?: number
+      hoverRadius?: number
+      hoverBorderWidth?: number
+    }
+    rectangle?: {
+      backgroundColor?: string
+      borderWidth?: number
+      borderColor?: string
+      borderSkipped?: string
+    }
+  }
+
+  interface Element {
+
+  }
+
+  class ChartInstance {
+    destroy(): void;
+    update(duration?: number, lazy?: boolean): void;
+    render(duration?: number, lazy?: boolean): void;
+    stop(): void;
+    resize(): void;
+    clear(): void;
+    toBase64Image(): string;
+    generateLegend(): string;
+    getElementAtEvent(e: Event): Element; // Check required. What does element return type contain? is Event correct here?
+    getElementsAtEvent(e: Event): Array<Element>; // Check required
+    getDatasetAtEvent(e: Event): Array<Element>; // Check required
+  }
+
+  interface LineChartData extends ChartData {
+    datasets?: Array<ChartDataSet & {
+      label?: string
+      fill?: boolean
+      backgroundColor?: string
+      borderColor?: string | Array<string>
+      borderCapStyle?: string
+      borderDash?: Array<number>
+      borderDashOffset?: number
+      borderJoinStyle?: string
+      pointBorderColor?: string | Array<string>
+      pointBackgroundColor?: string | Array<string>
+      pointBorderWidth?: number | Array<number>
+      pointHoverRadius?: number | Array<number>
+      pointHoverBackgroundColor?: string | Array<string>
+      pointHoverBorderColor?: string | Array<string>
+      pointHoverBorderWidth?: number | Array<number>
+      tension?: number
+      yAxisID?: string
+    }>
+  }
+
+  interface LineChartOptions extends ChartOptions {
+    showLines?: boolean
+    stacked?: boolean
+    scales?: {
+      xAxes?: Array<ScaleOptions & {
+        id?: string
+      }>
+      yAxes?: Array<ScaleOptions & {
+        id?: string
+      }>
+    }
+  }
+
+  class LineChartInstance extends ChartInstance {
+    data: LineChartData;
+    options: LineChartOptions;
+  }
+
+  interface BarChartData extends ChartData {
+    datasets?: Array<ChartDataSet & {
+      label?: string
+      backgroundColor?: string | Array<string>
+      borderColor?: string | Array<string>
+      borderWidth?: number | Array<number>
+      hoverBackgroundColor?: string | Array<string>
+      hoverBorderColor?: string | Array<string>
+      yAxisID?: string
+    }>
+  }
+
+  interface BarChartOptions extends ChartOptions {
+    stacked?: boolean
+    scales?: Array<{
+      xAxes?: Array<ScaleOptions & {
+        position?: string
+        id?: string
+        categoryPercentage?: number
+        barPercentage?: number
+        gridLines?: Array<ScaleGridLinesOptions & {
+          offsetGridLines?: boolean
+        }>
+      }>
+      yAxes?: Array<ScaleOptions & {
+        position?: string
+        id?: string
+      }>
+    }>
+  }
+
+  class BarChartInstance extends ChartInstance {
+    data: BarChartInstance;
+    options: BarChartOptions;
+  }
+
+  interface RadarChartData extends ChartData {
```

Comment:
I think this could extend LineChartData. The radar chart supports `tension`, `borderWidth`, `fill`, `borderCapStyle`, `borderDash`, `borderDashOffset`, `borderJoinStyle`